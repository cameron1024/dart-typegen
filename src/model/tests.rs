use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
