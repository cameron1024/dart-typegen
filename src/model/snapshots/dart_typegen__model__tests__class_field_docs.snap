---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    meta: None,
    defaults: None,
    enums: [],
    classes: [
        Class {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Foo",
                span: SourceSpan {
                    offset: SourceOffset(
                        45,
                    ),
                    length: 5,
                },
            },
            allow_non_pascal_case: None,
            fields: [
                Field {
                    name: SpannedScalar {
                        value: "x",
                        span: SourceSpan {
                            offset: SourceOffset(
                                61,
                            ),
                            length: 3,
                        },
                    },
                    ty: SpannedScalar {
                        value: "String",
                        span: SourceSpan {
                            offset: SourceOffset(
                                70,
                            ),
                            length: 8,
                        },
                    },
                    defaults_to: None,
                    defaults_to_dart: None,
                    docs: Some(
                        SpannedScalar {
                            value: "single line doc",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    90,
                                ),
                                length: 17,
                            },
                        },
                    ),
                    to_json: None,
                    from_json: None,
                    json_key: None,
                },
                Field {
                    name: SpannedScalar {
                        value: "y",
                        span: SourceSpan {
                            offset: SourceOffset(
                                121,
                            ),
                            length: 3,
                        },
                    },
                    ty: SpannedScalar {
                        value: "String",
                        span: SourceSpan {
                            offset: SourceOffset(
                                130,
                            ),
                            length: 8,
                        },
                    },
                    defaults_to: None,
                    defaults_to_dart: None,
                    docs: Some(
                        SpannedScalar {
                            value: "\nA multi-line doc\n\nSome more stuff\n    ",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    150,
                                ),
                                length: 44,
                            },
                        },
                    ),
                    to_json: None,
                    from_json: None,
                    json_key: None,
                },
            ],
            docs: None,
            json_key_case: None,
            json_discriminant_value: None,
            annotations: None,
            builder_annotations: None,
            extra_dart: None,
            builder_extra_dart: None,
        },
    ],
    unions: [],
}
